### DISCUSSION TITLE
Diagnostics

### DISCUSSION THREAD

**Discussion by granthusbands** [porsager/postgres#760] [Q&A] [answered]
Is there any way to get useful diagnostics while building queries? I've got a large constructed query and when I await it or await the .describe(), I just get an exception `PostgresError: syntax error at or near "$29"` and the query object is a weirdly nested and convoluted thing made of query fragments that aren't flattened. I just want to see the flattened query that it tries to actually run, to have _any_ idea of what is actually going wrong.

**Comment by granthusbands** [comment, marked as answer]
Okay, I've found the answer - catch the exception and look at the `query` and `parameters` properties of it. If I find time, I'll submit a pull request for that to be documented.

**Reply by ghost** [reply]
Is [Error Handling](https://github.com/porsager/postgres#error-handling) what you're looking for?

**Reply by granthusbands** [reply]
That helps, thanks, and it would have helped my searches if it mentioned the word "exception". The `query` property not being enumerable is a surprising choice and is part of what delayed me. But in the first instance, before I had the errors, I had a query fragment that I'd built using nested fragments and just wanted to see what the resultant query text would be. I didn't want to actually send it to the database, but that appears to be the only way.

**Reply by ghost** [reply]
[Query descriptions](https://github.com/porsager/postgres#query-descriptions) may be useful for you.

**Reply by granthusbands** [reply]
I'd tried that and got similar syntax error exceptions (that I didn't know at the time how to get anything useful from); the query is still being sent to the database, so the documentation is potentially misleading. What would really help in examining query fragments is to be able to flatten them without executing them or somehow get the query text without involving the server.

**Reply by ghost** [reply]
I looked briefly but did not see any issues created for this feature. I think having this feature can be useful for debugging. I think you might want to consider creating a new issue.

**Reply by wms** [reply]
I've raised an issue to support a better way of doing this at https://github.com/porsager/postgres/issues/816
